api_request.go

```diff
@@ -39,40 +38,52 @@ func newDeadlineTransport(timeout time.Duration) *http.Transport {
 	return transport
 }
 
-func apiRequestNegotiateV1(method string, endpoint string, body io.Reader) (*simplejson.Json, error) {
+type generalResp struct {
+	Status     string      `json:"status_txt"`
+	StatusCode int         `json:"statuc_code"`
+	Data       interface{} `json:"data"`
+}
+
+func apiRequestNegotiateV1(method string, endpoint string, body io.Reader, ret interface{}) error {
```

Comment:
I would just add a line of documentation that this function "stores the result in the value pointed to by ret" or something along those lines, the same way that json.Unmarshal is documented as "Unmarshal parses the JSON-encoded data and stores the result in the value pointed to by v."

That clarifies why the interface{} works in both cases.